hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -759,6 +769,120 @@ public static HoodieData<HoodieRecord> convertMetadataToColumnStatsRecords(Hoodi
         });
   }
 
+  static HoodieData<HoodieRecord> convertMetadataToRecordIndexRecords(HoodieEngineContext engineContext,
+                                                                      HoodieCommitMetadata commitMetadata,
+                                                                      HoodieMetadataConfig metadataConfig,
+                                                                      HoodieTableMetaClient dataTableMetaClient,
+                                                                      int writesFileIdEncoding,
```

Comment:
should this be enum

Reply:
method bit long, IMO. we need small methods that can be easily tested?

Reply:
sure. will split it up into smaller ones

Reply:
sorry, which one you meant for enum. 
for writesFileIdEncoding, we only have two possible values for now. 0 -> UUID based fileID. 1 -> random string.